### ISSUE TITLE
Some settings revert to older when opening settings dialog

### ISSUE THREAD

**Issue by softrabbit** [LMMS/lmms#1814]
```
$ grep -o 'mididev=".*"' ~/.lmmsrc.xml 
mididev="ALSA-Sequencer (Advanced Linux Sound Architecture)"
```
Start LMMS, change MIDI device in settings dialog, then:
```
$ grep -o 'mididev=".*"' ~/.lmmsrc.xml 
mididev="Dummy (no MIDI support)"
```
Open settings dialog, don't even look at the MIDI tab, just click "OK"
```
$ grep -o 'mididev=".*"' ~/.lmmsrc.xml 
mididev="ALSA-Sequencer (Advanced Linux Sound Architecture)"
```

Looks like this setting reverts to previously loaded data whenever the settings dialog is opened instead of what's been set (and saved!) previously. There might be other settings with similar behavior as well.

**Comment by M374LX** [comment]
The same happens with the audio device selection.

Clicking on the "OK" button on the settings dialog calls SetupDialog::accept(), which sets the value of "mididev" on CongigManager. Nevertheless, the value of "mididev" is never read within SetupDialog. Again, the same happens with "audiodev".

**Comment by tresf** [comment]
@softrabbit @mikobuntu Is this related at all to #1731?

**Comment by mikobuntu** [comment]
@tresf  I will check this soon to see if the bugs are related. Regarding the original posters @softrabbit  bug, I think this is working as it should e.g dummy midi is a fallback and once LMMS picks up ( on restart ) an available midi port it will connect, which usually is the last one that would have been chosen. To test change form Alsa-Seq-Midi to Alsa-Raw-Midi and the settings will stick . I can/will test this further to be absolutely sure, but this is my gut instinct on this one :)

**Comment by softrabbit** [comment]
@tresf, not related to #1731. That one's about MIDI connections, this is about MIDI driver selection in the setup dialog.

**Comment by tresf** [comment]
Closed via #1832
